client/pkg/circuitbreaker/circuit_breaker.go

```diff
@@ -309,3 +310,25 @@ func (s *State[T]) onResult(overloaded Overloading) {
 		panic("unknown state")
 	}
 }
+
+// Define context key type
+type cbCtxKey struct{}
+
+// Key used to store circuit breaker
+var CircuitBreakerKey = cbCtxKey{}
```

Comment:
It's just another way to do it. If it's complicated, I'm ok to keep the status quo.

Reply:
pros:
1. align with the retry mechanism
2. use the interceptor, the order is easier to control 
3. it can be changed to request level instead of client level
4. we can only modify TiDB to add a new type if we remove the generic
cons:
1. it's more complicated, each caller needs to maintain the circuit breaker
2. client level is easier to debug

Reply:
It may not be very convenient for the caller to switch during configuration changes.🤔

Reply:
Maybe we can get it directly from the SystemVar?

Reply:
We need switch it for every client.

Reply:
What if we use a global variable?

Reply:
What problem does the global variable resolve?🤔